### DISCUSSION TITLE
[Feature] Merge multiple faces

### DISCUSSION THREAD

**Discussion by eygraber** [immich-app/immich#3894] [Feature Request]
### The feature

I recently added 10k photos/videos to my library, and each individual person has had many faces generated for them. I've been going through each face and merging them one by one into the correct face (by setting the name, not by using the merge feature), but that's been very slow and tedious. 

I think it would be great if on the merge feature, I was able to select multiple other faces to merge into it. That way I could browse through the list of faces, selecting the matching ones, and then merge them all in at the end when I'm ready. 

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
This is already possible. On the main people page, select 'merge faces' on an entry, and then you can select multiple faces in the list to merge into it.

**Reply by eygraber** [reply]
Oh very cool, didn't notice that. Thanks!

**Reply by bmondream** [reply]
I wonder, why is it limited to only up to five faces? I can imagine it could burden the server if the merged faces were linked to thousands of photos, but why not leave that up to the user and perhaps display a warning?
https://github.com/immich-app/immich/blob/c6f56d959185ee59a0c3c1d864a657af1b5197f9/web/src/lib/components/faces-page/merge-face-selector.svelte#L58-L64

**Reply by bmondream** [reply]
So I managed to temporarily change it in the browser by editing the Javascript code snippet the browser downloads when you visit the merging face page. I did this in Brave, but the same approach should work in Chrome (and likely in any Chromium-based browser) as well. In case anyone reading this has the same issue, here's a guide to how I did that (_tested with Immich v1.90.2_):

1) navigate to the merge face page (select a person in the Explore tab, click `Merge person`)
2) open devtools (F12)
3) navigate to the `Sources` tab
4) find and open `_app/immutable/nodes/18.44ecc5a4.js` in the tree view on the left side (I'm not sure if the filename is static, YMMV)
5) press `Ctrl + F` and search for `You can only`
6) edit the condition surrounding the searched text (in my case `if (i.length >= 5)`) to whatever value you want
7) press `Ctrl + S` to save the changes
8) right-click the edited JS file and click `Override content`
9) click `Select folder` that showed up and choose a directory for the changed JS file (I think it's best to save it in an empty directory)
10) refresh Immich page

and then you can merge as many faces as you set.

![2023-12-22_01-54](https://github.com/immich-app/immich/assets/36151417/989c12c2-c840-4930-8d3c-8c8b66fd906c)

I'm aware there are more solutions, all probably more permanent than this, but I didn't need it more than once so far.
To make the change last longer, I believe you could `docker exec` into the `immich_server` container, find the mentioned JS file and edit it there. That should (I think) last until a new Immich update.

**Reply by lukyjay** [reply]
> 5. press `Ctrl + F` and search for `You can only`

In the current version you need to search all files for: merge_people_limit
Mine was in: 16.aqnNSgqQ.js

**Comment by masterT** [comment]
This is not possible with the mobile application.

**Reply by Supermankal718** [reply]
Yes I noticed it the other night that merging isn’t available on mobile. Would be great since auto loading is from our mobiles.

**Comment by supaeasy** [comment]
This solution used to work for me but now it somehow doesnt anymore. Is there no other way?

**Reply by lukyjay** [reply]
Check my reply above. I did this just last week
https://github.com/immich-app/immich/discussions/3894#discussioncomment-11657966

**Reply by supaeasy** [reply]
I figured it out again; had to reload the page after editing :-)
But I still don't unterstand why there is a Setting for EVERYTHING except something this much of a needless obstacle. I am in the process of merging 40k "Persons" and 5 limit was killing me...

**Comment by Maximus-CZ** [comment]
The 5 maximum limit is so arbitrary, useless and anoying. I needed to group faces from 5k photos and some people had 50+ faces detected over years.  (No, I couldnt adjust the parameter to make them group automatically because other, visually similar people were getting grouped incorrectly then).  

Thank god for being able to raise the limit at least in this hacky way, the extra 4-5 clicks needed after every merge to confirm and restart the merging process after 5 faces merged would otherwise be totally blocking me from proceeding. 

I dont see a single positive usecase for this, imo should be removed alltogether, or at least give user some normal way to bypass this arbitrary limit.

**Reply by supaeasy** [reply]
I have opened a PR for this issue, it was bugging me too hard. I hope this gets merged. https://github.com/immich-app/immich/pull/21671

**Reply by non-bin** [reply]
Seems like it was closed, any news?
